pages/battle.php

```diff
@@ -143,11 +143,21 @@ function processBattleFightEnd(BattleManager|BattleManagerV2 $battle, User $play
         );
         $result .= "You have earned $village_point_gain point for your village.[br]";
 
-        // Calculate rep gains
-        if ($player->reputation->canGain(false, true) && UserReputation::PVP_REP_ENABLED) {
-            $rep_gained = $player->reputation->handlePvPWin($player, $battle->opponent);
-            if ($rep_gained > 0) {
-                $result .= "You have earned $rep_gained village reputation.[br]";
+        if ($battle->retreat) {
+            // Calculate rep gains
+            if ($player->reputation->canGain(false, true) && UserReputation::PVP_REP_ENABLED) {
```

Comment:
```suggestion
            if ($player->reputation->canGain(check_mission_cd: false, check_pvp: true) && UserReputation::PVP_REP_ENABLED) {
```
This will make it much easier to figure out what's going on here instead of just seeing 
![image](https://github.com/elementum-games/shinobi-chronicles/assets/1691258/2f6903ca-a42b-4b5b-9008-8c342261cd86)